# Ejercicios sobre cadenas

<a href="https://colab.research.google.com/github/Ferorti/escuela2021/blob/main/ejercicios/cadenas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

No se olvide de ejecutar el siguiente código para tener las cadenas que utilizaremos en los ejercicios.

In [1]:
cadena1 = "  \t    GENE XXX tiene id YYY y ZZZ aa---"
cadena2 = "Gene\tUniprot\tAA\tPeso\nBRAF\tP15056\t766\t84436.135 DA\nEGFR\tP00533\t1210\t134276.185\nTP53\tP04637\t393\t43652.79 DA\nKRAS\tP01116\t189\t21655.67\n"
print(cadena1)
print(cadena2)

  	    GENE XXX tiene id YYY y ZZZ aa---
Gene	Uniprot	AA	Peso
BRAF	P15056	766	84436.135 DA
EGFR	P00533	1210	134276.185
TP53	P04637	393	43652.79 DA
KRAS	P01116	189	21655.67



## Ejercicio 1

Elimine los caracteres blancos del inicio de la cadena1 y los caracteres "-" del final de la cadena1. Imprima esta nueva cadena

In [2]:
# respuesta ejercicio 1

## Ejercicio 2

Imprima la cadena1 resultante del **Ejercicio 1**

*   toda en minúsculas
*   toda en mayúsculas
*   toda en minúsculas y solo con la primera letra en mayúscula



In [3]:
# respuesta ejercicio 2

## Ejercicio 3

Realice las siguientes sustituciones en la cadena1, e imprima la nueva cadena:
 
*   XXX por BRAF
*   YYY por P15056
*   ZZZ por 766



In [4]:
# respuesta ejercicio 3

## Ejercicio 4

Divida la cadena2 por el separador \t obteniendo y guarde el resultado

In [5]:
# respuesta ejercicio 4

## Ejercicio 5

Utilize el resultado del **Ejercicio 4** para unir los elementos de la lista en una nueva cadena que utilize como separador el caracter ","

In [6]:
# respuesta ejercicio 5

## Ejercicio 6

Convierta la cadena2 en una lista de listas como la siguiente:

```python
[
  ["BRAF", "P15056", 766, 84436.135], 
  ["EGFR", "P00533", 1210, 134276.185],
  ["TP53", "P04637", 393, 43652.79], 
  ["KRAS", "P01116", 189, 21655.67]
]
```

Sugerencias:

*   Utilize el método .split
*   Utilize ciclo for o while 





In [7]:
# respuesta ejercicio 6

## Ejercicio 7

Convierta la cadena2 en un diccionario como el siguiente:


```python
{
  "BRAF": {"uniprot": "P15056", "aa": 766, "peso": 84436.135}, 
  "EGFR": {"uniprot": "P00533", "aa": 1210, "peso": 134276.185}, 
  "TP53": {"uniprot": "P04637", "aa": 393,  "peso": 43652.79}, 
  "KRAS": {"uniprot": "P01116", "aa": 189, "peso": 21655.67}
}
```





Sugerencias:

*   Utilize él método split.
*   Utilize ciclo for o while.

In [8]:
# respuesta ejercicio 7


## Ejercicio 8

Haciendo uso de la lista obtenida en el **Ejercicio 6** imprima un texto por cada lista en ella que tenga el siguiente formato:

*   cada nombre de gene ocupe 8 espacios y esté alineado a la izquierda 
*   cada id de uniprot ocupe 10 espacios, esté alineado al centro y utilize como caracter de relleno el "*"
*   cada cantidad de aminoacidos ocupe 5 espacios y esté alineado a la derecha
*   después halla dos tabulaciones 
*   cada peso molecular ocupe 10 espacios, dos de los espacios reservados para los decimales y esté alineado a la derecha

Debe quedar una cadena como la siguiente:

`
BRAF    **P15056**  766		  84436.13
`

Sugerencia: Utilize la función format. Si le es más cómodo puede utilizar el diccionario obtenido en el **Ejercicio 7**.



In [9]:
# respuesta ejercicio 8